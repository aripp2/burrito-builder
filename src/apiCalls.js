export const getOrders = async () => {
  const response = await fetch('http://localhost:3001/api/v1/orders')
  if(!response.ok) {
    throw Error('Unable to get orders right now. Try again later.')
  }
  const orders = await response.json()
  return orders.orders
}

