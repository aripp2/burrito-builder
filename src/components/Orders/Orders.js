import React, { Component } from 'react';
import { connect } from 'react-redux';
import { setOrders } from '../../actions';
import { getOrders } from '../../apiCalls';

import './Orders.css';

export class Orders extends Component {
  // constructor() {
  //   super();
  //   this.state = {

  //   }
  // } 

  async componentDidMount() {
    const { setOrders } = this.props;
    try {
      const orders = await getOrders();
      setOrders(orders);
    } catch ({ message }) {
      console.log(message)
    }
  }

  render() {
    const { orders } = this.props
    const orderEls = orders.map(order => {
    return (
      <div className="order" key={order.id}>
        <h3>{order.name}</h3>
        <ul className="ingredient-list">
          {order.ingredients.map(ingredient => {
            return <li key={ingredient}>{ingredient}</li>
          })}
        </ul>
      </div>
    )
  });
    return (
      <section>
        { orderEls.length ? orderEls : <p>No orders yet!</p> }
      </section>
    )
  }
}

export const mapStateToProps = ({ orders }) => ({
  orders
});

export const mapDispatchToProps = dispatch => ({
  setOrders: orders => dispatch(setOrders(orders))
});

export default connect(mapStateToProps, mapDispatchToProps)(Orders);