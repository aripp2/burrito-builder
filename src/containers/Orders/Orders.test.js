import React from 'react';
import { shallow } from 'enzyme';
import { Orders, mapStateToProps, mapDispatchToProps } from './Orders';
import { getOrders } from '../../apiCalls';
import { setOrders } from '../../actions';

jest.mock('../../apiCalls.js');

const mockOrders = [
  {
    name: 'Amy',
    ingredients: [
      'steak',
      'pico de gallo',
      'guacamole'
    ],
    id: 4
  },
  {
    name: 'Raelyn',
    ingredients: [
      'queso fresco',
      'carnitas',
      'guacamole',
      'cilantro',
      'beans'
    ],
    id: 5
  }];
const mockSetOrders = jest.fn();

describe('Orders', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Orders 
      orders={mockOrders}
      setOrders={mockSetOrders}
      />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });

  it('should call getOrders', () => {
    getOrders.mockImplementation(() => Promise.resolve(mockOrders))
    expect(getOrders).toHaveBeenCalled();
  });

});

describe('mapStateToProps', () => {

  it('should return an object of orders array', () => {
    const mockState = {
      orders: mockOrders
    };
    const expected = {
      orders: mockOrders
    };
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });

});

describe('mapDispatchToProps', () => {

  const mockDispatch = jest.fn();

  it('calls dispatch with a setOrders action', () => {
    const actionToDispatch = setOrders(mockOrders);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.setOrders(mockOrders);
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  });

});