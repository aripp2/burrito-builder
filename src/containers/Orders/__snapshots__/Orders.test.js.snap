// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orders should match the snapshot 1`] = `
<section>
  <div
    className="order"
    key="4"
  >
    <h3>
      Amy
    </h3>
    <ul
      className="ingredient-list"
    >
      <li
        key="steak"
      >
        steak
      </li>
      <li
        key="pico de gallo"
      >
        pico de gallo
      </li>
      <li
        key="guacamole"
      >
        guacamole
      </li>
    </ul>
  </div>
  <div
    className="order"
    key="5"
  >
    <h3>
      Raelyn
    </h3>
    <ul
      className="ingredient-list"
    >
      <li
        key="queso fresco"
      >
        queso fresco
      </li>
      <li
        key="carnitas"
      >
        carnitas
      </li>
      <li
        key="guacamole"
      >
        guacamole
      </li>
      <li
        key="cilantro"
      >
        cilantro
      </li>
      <li
        key="beans"
      >
        beans
      </li>
    </ul>
  </div>
</section>
`;
